com.unity.perception/Runtime/Randomization/Randomizers/SampleRandomizers/Randomizers/ColorRandomizer.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using UnityEngine.Experimental.Perception.Randomization.Parameters;
+using UnityEngine.Experimental.Perception.Randomization.Randomizers.SampleRandomizers.Tags;
+
+namespace UnityEngine.Experimental.Perception.Randomization.Randomizers.SampleRandomizers
+{
+    /// <summary>
+    /// Randomizes the material color of objects tagged with a ColorRandomizerTag
+    /// </summary>
+    [Serializable]
+    [AddRandomizerMenu("Perception/Color Randomizer")]
+    public class ColorRandomizer : Randomizer
+    {
+        static readonly int k_BaseColor = Shader.PropertyToID("_BaseColor");
+
+        /// <summary>
+        /// Describes the range of random colors to assign to tagged objects
+        /// </summary>
+        public ColorHsvaParameter colorParameter;
+
+        /// <summary>
+        /// Randomizes the colors of tagged objects at the start of each scenario iteration
+        /// </summary>
+        protected override void OnIterationStart()
+        {
+            var taggedObjects = tagManager.Query<ColorRandomizerTag>();
```

Comment:
I agree that it might be helpful to associate randomizers with the tags they require through some type safe interface, but I think it is totally valid for a scenario to have a randomizer that requires a tag that no single existing GameObject has. The randomizer will simply accomplish nothing for the frame.

Reply:
I think Mohsen actually has a JIRA ticket considering this link between randomizers and the tags they use. I'll mark this review item as resolved and defer adding this functionality in a future PR.